manifests/datasource/oracle.pp

```diff
@@ -0,0 +1,36 @@
+# Private class.
+class keycloak::datasource::oracle (
+  $jar_source    = 'puppet:///modules/keycloak/database/oracle/ojdbc8.jar',
```

Comment:
If this jar can't be installed from an OS package then it may be better to default to undef and remove `ojdbc8.jar` from git tracking.  Would be better to document how someone can use a profile or some other logic to host their own jar source and define `$jar_source` to point to their local copy.

Reply:
Also only manage the `file` resource if `$jar_source` is defined to avoid trying to have Keycloak use empty jar file if parameter not defined.

Reply:
You are correct with the assumption, that other ojdbc driver can be used. I've commited updated changes with odjbc file and source now as optional module parameters and check is done in main class